### ISSUE TITLE
Fix FP S3516 (`no-invariant-returns`): returned value is changed between return statements

### ISSUE THREAD

**Issue by vilchik-elena** [SonarSource/SonarJS#1641]
```typescript
function createOpenTagMarkup(
  tagVerbatim: string,
  tagLowercase: string,
  props: Object,
  namespace: string,
  makeStaticMarkup: boolean,
  isRootElement: boolean,
): string {
  let ret = '<' + tagVerbatim;

  for (const propKey in props) {
    if (!hasOwnProperty.call(props, propKey)) {
      continue;
    }
    if (enableFlareAPI && propKey === 'listeners') {
      continue;
    }
    let propValue = props[propKey];
    if (propValue == null) {
      continue;
    }
    if (propKey === STYLE) {
      propValue = createMarkupForStyles(propValue);
    }
    let markup = null;
    if (isCustomComponent(tagLowercase, props)) {
      if (!RESERVED_PROPS.hasOwnProperty(propKey)) {
        markup = createMarkupForCustomAttribute(propKey, propValue);
      }
    } else {
      markup = createMarkupForProperty(propKey, propValue);
    }
    if (markup) {
      ret += ' ' + markup;
    }
  }

  // For static pages, no need to put React ID and checksum. Saves lots of
  // bytes.
  if (makeStaticMarkup) {
    return ret;
  }

  if (isRootElement) {
    ret += ' ' + createMarkupForRoot();
  }
  return ret;
}

```

`packages/react-dom/src/server/ReactPartialRenderer.js` in react

**Comment by ilia-kebets-sonarsource** [comment]
The code example above does not raise an issue.

**Comment by yassin-kammoun-sonarsource** [comment]
We can close it, but it would be nice to understand why.
